lib/backend/k8s/k8s.go

```diff
@@ -739,9 +741,85 @@ func (c *KubeClient) deleteGlobalConfig(k *model.KVPair) error {
 }
 
 func (c *KubeClient) getHostConfig(k model.HostConfigKey) (*model.KVPair, error) {
+	if k.Name == "IpInIpTunnelAddr" {
+		n, err := c.clientSet.Nodes().Get(k.Hostname, metav1.GetOptions{})
+		if err != nil {
+			return nil, resources.K8sErrorToCalico(err, k)
+		}
+
+		tunIp, err := getTunIp(n)
+		if err != nil {
+			return nil, err
+		}
+
+		return &model.KVPair{Key: k, Value: tunIp}, nil
+	}
+
 	return nil, errors.ErrorResourceDoesNotExist{Identifier: k}
 }
 
 func (c *KubeClient) listHostConfig(l model.HostConfigListOptions) ([]*model.KVPair, error) {
-	return []*model.KVPair{}, nil
+	var kvps = []*model.KVPair{}
+
+	// First see if we were handed a specific host, if not list all Nodes
+	if l.Hostname == "" {
+		nodes, err := c.clientSet.Nodes().List(v1.ListOptions{})
+		if err != nil {
+			return nil, resources.K8sErrorToCalico(err, l)
+		}
+
+		for _, node := range nodes.Items {
+			tunIp, err := getTunIp(&node)
+			if err != nil {
+				log.Warnf("Invalid IP for HostConfig: %s, %s", node.Name, node.Spec.PodCIDR)
+				return []*model.KVPair{}, nil
+			}
+
+			kvp := &model.KVPair{
+				Key: model.HostConfigKey{
+					Hostname: node.Name,
+					Name: "IpInIpTunnelAddr",
+				},
+				Value: tunIp,
+			}
+
+			kvps = append(kvps, kvp)
+		}
+	} else {
+		node, err := c.clientSet.Nodes().Get(l.Hostname, metav1.GetOptions{})
+		if err != nil {
+			return nil, resources.K8sErrorToCalico(err, l)
+		}
+
+		tunIp, err := getTunIp(node)
+		if err != nil {
+			return []*model.KVPair{}, nil
+		}
+
+		kvp := &model.KVPair{
+			Key: model.HostConfigKey{
+				Hostname: l.Hostname,
+				Name: "IpInIpTunnelAddr",
+			},
+			Value: tunIp,
+		}
+
+		kvps = append(kvps, kvp)
+	}
+
+	return kvps, nil
 }
+
+func getTunIp(n *v1.Node) (string, error) {
```

Comment:
If this returned a pointer to the KVPair instead of the string I think that would simplify a chunk of the code above.

If we hit an error parsing the CIDR then just return a nil KVPair and log.  I think for the list case above - you probably just want to skip over nodes that don't have a valid CIDR rather than returning no results at all.